lib/themes/dosomething/paraneue_dosomething/js/takeover/Takeover.js

```diff
@@ -89,55 +71,21 @@ class Takeover {
   }
 
   /**
-   * Binds click handlers to $button that sends an AJAX request and uses deferred promises to
-   * run callbacks in the order in which they are recieved.
-   *
-   * @param  {jQuery}   $button   Signup buttons
-   * @param  {object}   request   An object that defines the properties we need to make a request.
-   */
-  handleSubmit($button, request) {
-    // Create custom deferred promise object.
-    let ajaxPromise = $.Deferred().resolve().promise();
-
-    $button.click(e => {
-      e.preventDefault();
-
-      if (! this.$el.find('.validation__message.has-error').length) {
-        // Make the ajax call, store the promise.
-        const promise = this.sendRequest(request);
-
-        // Fire done call backs in the order in which they were recieved.
-        ajaxPromise = ajaxPromise.then(promise => { return promise } ).done( data => this.changeScreens('next'));
-      }
-    });
-  }
-
-  /**
    * Sends an AJAX request using the provided information in the request object that is passed in.
    *
    * @param  {request}   The object containing the details about the request i.e URL, method, data
    * @return {object}    The api response.
    */
   sendRequest(request) {
-    const _this = this;
-
-    return $.ajax({
+    fetch(request.url, {
       method: request.method,
-      url: request.url,
-      dataType: 'json',
+      credentials: 'same-origin',
       headers: {
-        'Accepts': 'application/json',
-        'Content-Type': 'application/json',
-        'X-CSRF-Token': $('meta[name="csrf-token"]').attr('content'),
-      },
-      data: request.data,
-      beforeSend: function() {
-        _this.toggleSpinner(request.button);
-      },
-      complete: function() {
-        _this.toggleSpinner(request.button);
+        'X-CSRF-Token': $('meta[name="csrf-token"]').attr('content')
       }
-    });
+    })
+    .then(response => response.json())
+    .then(json => request.done(json));
```

Comment:
Instead of passing a callback in the `request` parameter here, you can return the Promise from `fetch()` and continue chaining `.then()` on it. So you'd be able to call this method like:

```js
this.sendRequest({
  url    : `${this.postalCodeUrl}?postal_code=${zipCode}`,
  method : 'GET'
}).then(json => {
  // ...
});
```